src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2051,14 +2051,15 @@ internal override bool FilterAndAppendAndFreeDiagnostics(DiagnosticBag accumulat
         /// Filter out warnings based on the compiler options (/nowarn, /warn and /warnaserror) and the pragma warning directives.
         /// </summary>
         /// <returns>True when there is no error.</returns>
-        private bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<Diagnostic> incoming)
+        private bool FilterAndAppendDiagnostics(DiagnosticBag accumulator, IEnumerable<Diagnostic> incoming, bool includeDiagnosticsWithSourceSuppression = false)
         {
             bool hasError = false;
 
             foreach (Diagnostic d in incoming)
             {
                 var filtered = _options.FilterDiagnostic(d);
-                if (filtered == null)
+                if (filtered == null ||
+                    (!includeDiagnosticsWithSourceSuppression && filtered.HasSourceSuppression))
```

Comment:
Basically suppression in source is a way of baselining specific instances of issues, and suppress through compilation options is act of suppressing the rule across the compilation. There is indeed a distinction between the two and we never surfaced it. Now we make it clear: you can suppress rules for entire compilation and they will be dropped on the floor, and you can suppress specific instances, which you can fetch if you are interested.